packcheck.sh

```diff
@@ -1801,40 +1818,45 @@ show_build_env
 export HOME=$(echo ~)
 set_os_specific_vars # depends on HOME
 
-# Set path for installed utilities, e.g. stack, cabal, hpc-coveralls
-echo
-echo "Adding usual utility locations to PATH [$PATH]..."
-if test "$BUILD" = "cabal-v2"
-then
-  export PATH=$OS_APP_HOME/$OS_CABAL_DIR/bin:$PATH
-fi
-export PATH=$OS_APP_HOME/$OS_LOCAL_DIR/bin:$PATH
-echo
-echo "PATH is now set to [$PATH]"
 
-verify_build_config
+if test "$BUILD" = "hlint"
+then
 
-# if we are running from a stack environment remove GHC_PACKAGE_PATH so that
-# cabal does not complain
-# XXX this should be done from outside via env
-unset GHC_PACKAGE_PATH
-# stack does not work well with empty STACK_YAML env var
-test -n "$STACK_YAML" || unset STACK_YAML
+    # if we are running from a stack environment remove GHC_PACKAGE_PATH so that
+    # cabal does not complain
+    # XXX this should be done from outside via env
+    unset GHC_PACKAGE_PATH
+    # stack does not work well with empty STACK_YAML env var
+    test -n "$STACK_YAML" || unset STACK_YAML
 
-if test -n "$HLINT_COMMANDS" -o -n "$HLINT_TARGETS"
-then
-  if test -z "$(which_cmd hlint)"
-  then
-    if test -n "$HLINT_BUILD"
+    if test -z "$(which_cmd hlint)"
     then
-      build_hlint
-    else
-      install_hlint
+        if test -n "$HLINT_BUILD"
+        then
+            build_hlint
+        else
+            install_hlint
+        fi
     fi
-  fi
-  run_hlint
+    run_hlint
+
 else
-  build_compile
+
+    # Set path for installed utilities, e.g. stack, cabal, hpc-coveralls
+    echo
+    echo "Adding usual utility locations to PATH [$PATH]..."
+    if test "$BUILD" = "cabal-v2"
+    then
+        export PATH=$OS_APP_HOME/$OS_CABAL_DIR/bin:$PATH
+    fi
+    export PATH=$OS_APP_HOME/$OS_LOCAL_DIR/bin:$PATH
+    echo
```

Comment:
Earlier this was being done for hlint case as well. Don't we need this for hlint?

Reply:
We do.